code/atom.dm

```diff
@@ -1417,3 +1417,11 @@ TYPEINFO(/atom/movable)
 		if("icon_state")
 			src.icon_state = oldval
 			src.set_icon_state(newval)
+
+/atom/movable/proc/is_that_in_this(atom/movable/target)
+	if (target.loc == src)
+		return TRUE
+	for(var/atom/movable/AM in src)
+		if(AM.contents.len && AM.is_that_in_this(target))
+			return TRUE
```

Comment:
Might be more performant to recursively walk up the contents tree instead of iterating over all contents?

```suggestion
	if (istype(target.loc, /atom/movable))
	    return src.is_that_in_this(target.loc)
```